frame/staking/src/lib.rs

```diff
@@ -2063,74 +2081,85 @@ decl_module! {
 		///
 		/// # <weight>
-		/// E: number of edges. m: size of winner committee. n: number of nominators. d: edge degree
-		/// (16 for now) v: number of on-chain validator candidates.
-		///
-		/// NOTE: given a solution which is reduced, we can enable a new check the ensure `|E| < n +
-		/// m`. We don't do this _yet_, but our offchain worker code executes it nonetheless.
-		///
-		/// major steps (all done in `check_and_replace_solution`):
-		///
-		/// - Storage: O(1) read `ElectionStatus`.
-		/// - Storage: O(1) read `PhragmenScore`.
-		/// - Storage: O(1) read `ValidatorCount`.
-		/// - Storage: O(1) length read from `SnapshotValidators`.
-		///
-		/// - Storage: O(v) reads of `AccountId` to fetch `snapshot_validators`.
-		/// - Memory: O(m) iterations to map winner index to validator id.
-		/// - Storage: O(n) reads `AccountId` to fetch `snapshot_nominators`.
-		/// - Memory: O(n + m) reads to map index to `AccountId` for un-compact.
-		///
-		/// - Storage: O(e) accountid reads from `Nomination` to read correct nominations.
-		/// - Storage: O(e) calls into `slashable_balance_of_vote_weight` to convert ratio to staked.
-		///
-		/// - Memory: build_support_map. O(e).
-		/// - Memory: evaluate_support: O(E).
-		///
-		/// - Storage: O(e) writes to `QueuedElected`.
-		/// - Storage: O(1) write to `QueuedScore`
-		///
-		/// The weight of this call is 1/10th of the blocks total weight.
+		/// All weight notes are pertaining to the case of a better solution, in which we execute
+		/// the longest code path.
+		/// Weight: 0 + (35 μs * v) + (25 μs * n)
+		/// State reads:
+		/// 	- Initial checks:
+		/// 		- ElectionState, CurrentEra, QueuedScore
+		/// 		- SnapshotValidators.len()
+		/// 		- ValidatorCount
+		/// 		- SnapshotValidators
+		/// 		- SnapshotNominators
+		/// 	- Iterate over nominators:
+		/// 		- compact.len() * Nominators(who)
+		/// 		- (compact.len() - winners.len()) * avg_edge_count * SlashingSpans
+		/// 	- For `assignment_ratio_to_staked`: Basically read the staked value of each stash.
+		/// 		- (winners.len() + compact.len()) * (Ledger + Bonded)
+		/// 		- TotalIssuance (read a gzillion times potentially, but well it is cached.)
+		/// - State writes:
+		/// 	- QueuedElected, QueuedScore
 		/// # </weight>
-		#[weight = 100_000_000_000]
+		#[weight =
+			(35 * WEIGHT_PER_MICROS * (size.validators as Weight))
+			.saturating_add(25 * WEIGHT_PER_MICROS * (size.validators as Weight))
+			.saturating_add(T::DbWeight::get().reads(7))
+			.saturating_add(T::DbWeight::get().reads(compact.len() as Weight)) // Nominators
+			.saturating_add( // SlashingSpans
+				T::DbWeight::get().reads(
+					compact.len()
+						.saturating_sub(winners.len())
+						.saturating_mul(compact.average_edge_count())
+						as Weight
+				)
+			)
+			.saturating_add(T::DbWeight::get().reads(2 * ((winners.len() + compact.len()) as Weight)))
+			.saturating_add(T::DbWeight::get().reads(1))
+			.saturating_add(T::DbWeight::get().writes(2))
```

Comment:
same question: which reads and writes

Reply:
all same order as comments.

Reply:
I also tried to make things cleared, should be fine now